File: 프로그래머스/unrated/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(string myString, string pat) {
    int cnt = 0;
    
    for (int i = 0; i < myString.size(); i++)
    {
        int idx = myString.find(pat, i);
        if (idx != string::npos)
        {
            cnt++;
            i = idx;
        }    
    }
    
    
    return cnt;
}

File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.java
class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
        
        for (int i = 0; i <= myString.length() - pat.length(); i++) {
            if (myString.substring(i).startsWith(pat)) {
                answer++;
            }
        }
        
        return answer;
    }
}